CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -962,6 +1078,51 @@ local function CreateMobileHideChatIcon()
 	return CreateMenuItem(chatHideIconButton)
 end
 
+
+local function Create3DChatIcon(topBarInstance, panel)
+	local menuItem = CreateChatIcon(topBarInstance)
+
+
+	rawset(menuItem, "Hover", function(self, hovering)
+		if hovering then
+			self:SetImage("rbxasset://textures/ui/Chat/ChatDown.png")
+		else
+			self:SetImage("rbxasset://textures/ui/Chat/Chat.png")
+		end
+	end)
+
+	local function On3DInput(actionName, state, inputObj)
+		if state == Enum.UserInputState.Begin then
+			menuItem:ToggleChat()
+		end
+	end
+
+	local eaterAction = game:GetService("HttpService"):GenerateGUID()
+	local function EnableChat3DInput(enable)
+		if enable then
+			ContextActionService:BindCoreAction("ChatIcon3DInput", On3DInput, false, Enum.KeyCode.Space, Enum.KeyCode.ButtonA)
+			ContextActionService:BindAction(eaterAction, function() end, false, Enum.KeyCode.Space, Enum.KeyCode.ButtonA)
```

Comment:
I believe this may be unnecessary soon given my changes for gaze select. Once we confirm that works without a hitch I think we could just remove this bit entirely.